### PR TITLE
[docs] Add autogenerated CLI reference

### PR DESCRIPTION
**Pull request by abiogenesis-now**
@jessicayuen @hausdorff PTAL

Copied from https://github.com/ksonnet/kubecfg/pull/130:

> I figured it might be useful to have autogenerated markdown docs for the CLI that users can peruse in addition to using --help in the CLI itself. Let me know if you think this is unnecessary though.

(`./docs/cli-reference` can also be incorporated as a Git submodule into the ksonnet website repo---granted with a lot of formatting changes :P) 

We should edit the actual CLI command descriptions at some point, but I'm holding off on that for now since some of them are still WIP.

### CODE REVIEW

**Comment by abiogenesis-now** [comment]
not sure why this is erroring on the first build and not the second (it looks fine locally), will look into it later

**Comment by abiogenesis-now** [comment]
Updated, it seems like rebasing on the cgo -> go-jsonnet changes fixed the compilation issues

**Comment by jessicayuen** [comment]
I think we should run a re-gen after we merge @hausdorff's fixes for #50 and #38, then merge this.

**Comment by abiogenesis-now** [comment]
yup, will prob do that tomorrow morning (assuming fixes have been merged by then)

afterwards i'll do a pass to edit words

**Comment by abiogenesis-now** [comment]
@jessicayuen @hausdorff updated, PTAL

File: docs/generate/ks.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package main
+
+import (
+	"log"
+  "os"
```
Comment (jessicayuen) [inline docs/generate/ks.go]: nit: misformatted

File: docs/generate/ks.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+	"github.com/spf13/cobra/doc"
+	"github.com/ksonnet/ksonnet/cmd"
+)
+
+func main() {
+	outputDir := os.Args[1]
+
+  cmd := cmd.RootCmd
```
Comment (jessicayuen) [inline docs/generate/ks.go]: nit: misformatted

**Pull request merged**